brainwidemap/decoding/functions/process_motors.py

```diff
@@ -0,0 +1,386 @@
+        
+        # skip trial if any key info is nan for subsequ. trial    
+        if any(np.isnan([trials[k][tr + 1] for k in evts])):
+            continue            
+
+        # skip trial if too long
+        if trials['feedback_times'][tr] - trials['stimOn_times'][tr] > 10: 
+            continue   
+        
+        # skip block boundary trials
+        if trials['probabilityLeft'][tr] != trials['probabilityLeft'][tr+1]:
+            continue
+        '''
+        
+        a = wheelMoves['intervals'][:,1]
+        '''
+        b = trials['stimOn_times'][tr]
+        c = trials['feedback_times'][tr - 1]     
+        # making sure the motion onset time is in a coupled interval
+        ind = np.where((a < b) & (a > c))[0]
+        try:
+            a = a[ind][-1]
+        except:
+            continue   
+        
+        
+        difs.append(b-a)
+        '''
+
+        if stim_to_stim:
+            start_t = trials['stimOn_times'][tr]    
+                    
+        elif align == 'wheel_stop':            
+            start_t = a + lag    
+            
+        else:                                
+            start_t = trials[align][tr] + lag     
+                
+        if np.isnan(trials['contrastLeft'][tr]):
+            cont = trials['contrastRight'][tr]            
+            side = 0  # right side stimulus
+        else:   
+            cont = trials['contrastLeft'][tr]         
+            side = 1  # left side stimulus                   
+                  
+        sides.append(side) 
+        
+        if endTrial:
+            choices.append(trials['choice'][tr+1])
+        else:                              
+            choices.append(trials['choice'][tr])   
+                 
+        pleft.append(trials['probabilityLeft'][tr])
+         
+        for be in behaves: 
+            times = behaves[be][0] 
+            series = behaves[be][1] 
+            start_idx = find_nearest(times,start_t)        
+            if stim_to_stim:  
+                end_idx = find_nearest(times, trials['stimOn_times'][tr + 1])
+            else:
+                if sr[be] == 'T_BIN':
+                    end_idx = start_idx + int(duration/T_BIN)
+                else:
+                    fs = sr[be]
+                    end_idx = start_idx + int(duration*fs)              
+            
+            if (pawex and ('paw' in be)): #for illustration on frame
+                D[be].append([series[0][start_idx:end_idx],
+                              series[1][start_idx:end_idx]])            
+            else:  
+                '''  # bug inducing 
+                if start_idx > len(series):
+                    print('start_idx > len(series)')
+                    break      
+                '''  
+                D[be].append(series[start_idx:end_idx])         
+                  
+        T.append(tr)
+        kk+=1
+
+    print(kk, 'trials used')
+    return(D)
+
+
+
+def preprocess_motors_old(eid,metadata):
```

Comment:
This function is only used in preprocess_motors_old which is itself no longer used. Remove